packages/medusa-js/src/resources/admin/inventory-item.ts

```diff
@@ -0,0 +1,155 @@
+
+    if (query) {
+      const queryString = qs.stringify(query)
+      path += `?${queryString}`
+    }
+
+    return this.client.request("GET", path, undefined, {}, customHeaders)
+  }
+
+  /**
+   * Update an Inventory Item
+   * @experimental This feature is under development and may change in the future.
+   * To use this feature please install @medusajs/inventory
+   * @description updates an Inventory Item
+   * @returns the updated Inventory Item
+   */
+  update(
+    inventoryItemId: string,
+    payload: AdminPostInventoryItemsInventoryItemReq,
+    query?: AdminGetInventoryItemsItemParams,
+    customHeaders: Record<string, any> = {}
+  ): ResponsePromise<AdminInventoryItemsRes> {
+    let path = `/admin/inventory-items/${inventoryItemId}`
+
+    if (query) {
+      const queryString = qs.stringify(query)
+      path += `?${queryString}`
+    }
+
+    return this.client.request("POST", path, payload, {}, customHeaders)
+  }
+
+  /**
+   * Delete an Inventory Item
+   * @experimental This feature is under development and may change in the future.
+   * To use this feature please install @medusajs/inventory
+   * @description deletes an Inventory Item
+   * @returns the deleted Inventory Item
+   */
+  delete(
+    inventoryItemId: string,
+    customHeaders: Record<string, any> = {}
+  ): ResponsePromise<AdminInventoryItemsDeleteRes> {
+    const path = `/admin/inventory-items/${inventoryItemId}`
+    return this.client.request("DELETE", path, undefined, {}, customHeaders)
+  }
+
+  /**
+   * Retrieve a list of Inventory Items
+   * @experimental This feature is under development and may change in the future.
+   * To use this feature please install @medusajs/inventory
+   * @description Retrieve a list of Inventory Items
+   * @returns the list of Inventory Items as well as the pagination properties
+   */
+  list(
+    query?: AdminGetInventoryItemsParams,
+    customHeaders: Record<string, any> = {}
+  ): ResponsePromise<AdminInventoryItemsListWithVariantsAndLocationLevelsRes> {
+    let path = `/admin/inventory-items`
+
+    if (query) {
+      const queryString = qs.stringify(query)
+      path += `?${queryString}`
+    }
+
+    return this.client.request("GET", path, undefined, {}, customHeaders)
+  }
+
+  /**
+   * Update an Inventory Item
+   * @experimental This feature is under development and may change in the future.
+   * To use this feature please install @medusajs/inventory
+   * @description updates an Inventory Item
+   * @returns the updated Inventory Item
+   */
+  updateLocationLevel(
+    inventoryItemId: string,
+    locationId: string,
+    payload: AdminPostInventoryItemsItemLocationLevelsLevelReq,
+    customHeaders: Record<string, any> = {}
+  ): ResponsePromise<AdminInventoryItemsRes> {
+    const path = `/admin/inventory-items/${inventoryItemId}/location-levels/${locationId}`
+    return this.client.request("POST", path, payload, {}, customHeaders)
```

Comment:
todo: we are still missing the query params here, as part of the validator and OAS they are available